### PR TITLE
Use mio v0.8 in apps and tools

### PR DESCRIPTION
**Pull request by sile**
This is a follow-up PR of #1200 and updates the mio version to the latest one (v0.8).

### CODE REVIEW

File: apps/src/bin/quiche-server.rs [hunk 1]
```diff
@@ -61,22 +61,19 @@ fn main() {
     let args = ServerArgs::with_docopt(&docopt);
 
     // Setup the event loop.
-    let poll = mio::Poll::new().unwrap();
+    let mut poll = mio::Poll::new().unwrap();
     let mut events = mio::Events::with_capacity(1024);
 
     // Create the UDP listening socket, and register it with the event loop.
     let socket = net::UdpSocket::bind(args.listen).unwrap();
+    socket.set_nonblocking(true).unwrap();
```
Comment (junhochoi) [inline apps/src/bin/quiche-server.rs:69]: Do we need this?

File: apps/src/bin/quiche-server.rs [hunk 2]
```diff
@@ -61,22 +61,19 @@ fn main() {
...
 
-    let socket = mio::net::UdpSocket::from_socket(socket).unwrap();
-    poll.register(
-        &socket,
-        mio::Token(0),
-        mio::Ready::readable(),
-        mio::PollOpt::edge(),
-    )
-    .unwrap();
+    let mut socket = mio::net::UdpSocket::from_std(socket);
```
Comment (junhochoi) [inline apps/src/bin/quiche-server.rs:73]: I think we can merge L68 (calling `bind()`) and this line into calling `mio::net::UdpSocket::bind()` once.

**Review by junhochoi — CHANGES_REQUESTED** [review]

File: apps/src/bin/quiche-server.rs (hunk 1 above)
Comment (sile) [reply apps/src/bin/quiche-server.rs:69]: Without this line, the socket will blocks on the recv_from call below (as far as I confirmed in my local env): https://github.com/cloudflare/quiche/blob/0107fb90694d0144bcf76a125c67abb84e2fe204/apps/src/bin/quiche-server.rs#L170

Please also see the comment in the mio repo: https://github.com/tokio-rs/mio/blob/b2bbc1029849c939031fb396475d1a19da9786a6/tests/udp_socket.rs#L102-L105

File: apps/src/bin/quiche-server.rs (hunk 2 above)
Comment (sile) [reply apps/src/bin/quiche-server.rs:73]: Sounds good, thanks!
The result of `mio::net::UdpSocket::bind()` seems a non-blocking socket by default, so we can eliminate the `set_nonblocking` call too.

**Review by sile — COMMENTED** [review]
@junhochoi Thank you for your review. I will reflect your commets soon.

File: apps/src/bin/quiche-server.rs (hunk 2 above)
Comment (sile) [reply apps/src/bin/quiche-server.rs]: Fixed in https://github.com/cloudflare/quiche/pull/1202/commits/935e8d6ac79340a4172da931cd90f66ec963ea67
Note that I also updated the UDP socket creation code merged in the previous PR (#1200).

**Review by junhochoi — APPROVED** [review]
LGTM

**Comment by ghedo** [comment]
This is merged now, thanks @sile!

**Comment by sile** [comment]
🎉

**Pull request closed without merging**